python/taichi/ui/staging_buffer.py

```diff
@@ -78,6 +79,29 @@ def copy_all_to_vbo(
                 vbo[i][8:12] = color[i]
 
 
+@kernel
+def copy_all_to_vbo_particle(  # The vbo copy function used for vertices of particles and circles
+    vbo: ti.types.ndarray(element_dim=1),
```

Comment:
Since it's inside Taichi's implementation, it's better to import this like 

```
import taichi.types.ndarray_type.ndarray as ndarray_type
```

And then:
```
vbo: ndarray_type(element_dim=1),
```